From 79bff679d6f8b5b1895c8a502c0fb4e83a5f0c5b Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Tue, 31 Mar 2015 17:27:13 +0800
Subject: [PATCH 337/338] gmin: ov2722 + ov5693 add power resource config

This fixed the bug that power rails are still on after S3.

Change-Id: I52c5be25b1c6c3e63f7804ba920693f52fbc1d23
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-28141
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/49022
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Approver: Bingbu Cao <bingbu.cao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Jing Tao <jing.tao@intel.com>
Build: IMIN-legacy (SC)
Integrator: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov2722.c        | 8 +++++++-
 .../external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c | 7 ++++++-
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
index 41cd21c..486557c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov2722.c
@@ -1447,6 +1447,9 @@ static int ov2722_probe(struct i2c_client *client,
 	struct ov2722_device *dev;
 	void *ovpdev;
 	int ret;
+#ifdef CONFIG_GMIN_INTEL_MID
+	struct acpi_device *adev;
+#endif
 
 	dev = kzalloc(sizeof(*dev), GFP_KERNEL);
 	if (!dev) {
@@ -1461,10 +1464,13 @@ static int ov2722_probe(struct i2c_client *client,
 
 	ovpdev = client->dev.platform_data;
 #ifdef CONFIG_GMIN_INTEL_MID
-	if (ACPI_COMPANION(&client->dev))
+	adev = ACPI_COMPANION(&client->dev);
+	if (adev) {
+		adev->power.flags.power_resources = 0;
 		ovpdev = gmin_camera_platform_data(&dev->sd,
 						   ATOMISP_INPUT_FORMAT_RAW_10,
 						   atomisp_bayer_order_grbg);
+	}
 #endif
 	ret = ov2722_s_config(&dev->sd, client->irq, ovpdev);
 	if (ret)
diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
index 26d3c7f..3ca28dc 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.c
@@ -2088,6 +2088,7 @@ static int ov5693_probe(struct i2c_client *client,
 	int i2c;
 	int ret = 0;
 	void *pdata = client->dev.platform_data;
+	struct acpi_device *adev;
 
 	/* Firmware workaround: Some modules use a "secondary default"
 	 * address of 0x10 which doesn't appear on schematics, and
@@ -2112,10 +2113,14 @@ static int ov5693_probe(struct i2c_client *client,
 	dev->fmt_idx = 0;
 	v4l2_i2c_subdev_init(&(dev->sd), client, &ov5693_ops);
 
-	if (ACPI_COMPANION(&client->dev))
+	adev = ACPI_COMPANION(&client->dev);
+	if (adev) {
+		adev->power.flags.power_resources = 0;
 		pdata = gmin_camera_platform_data(&dev->sd,
 						  ATOMISP_INPUT_FORMAT_RAW_10,
 						  atomisp_bayer_order_bggr);
+	}
+
 	if (!pdata)
 		goto out_free;
 
-- 
1.9.1

